### DISCUSSION TITLE
Hi, is there any way to implement the on_submit event in the form with html tags? :)

### DISCUSSION THREAD

**Discussion by cllatser** [reflex-dev/reflex#2768] [Q&A] [answered]
Hi, as you know I use Reflex with HTML tags and Tailwind. Now I find that the <form> tag does not have the "on_submit" option. What would be the correct way to implement it if possible?

Thank you very much.

```
@rx.page(title="jwt_demo", on_load=FormInputState.obtain_token())
def index() -> rx.Component:
    return rx.el.div(
        rx.el.form(
            rx.el.input(
                name="email",
                default_value="search",
                placeholder="Input mail here...",
                type="email",
                required=True,
            ),
            rx.el.input(
                name="password",
                default_value="search",
                placeholder="Input password here...",
                type="password",
                required=True,
            ),
            rx.el.button(
                "Submit",
                type="submit",
                class_name="w-full"),
            **on_submit = ??????????**
            class_name="flex flex-col"
        ),
        reset_on_submit=True,
        class_name="min-h-screen flex flex-col justify-center items-center"
    )


app = rx.App()
```

**Comment by masenf** [comment]
I think it's not supported at the moment... one workaround would be to set the form `action` to a route and let the browser submit the form traditionally.

I think it probably would make sense to move the `on_submit` machinery that was implemented for the radix form in the base HTML form element.

**Reply by cllatser** [reply]
I have the first option considered, but it would be fantastic if the action were implemented as in the Radix-UI form. There's no rush, whenever it's possible and can be looked into, would be fantastic.

**Comment by masenf** [comment, marked as answer]
#2776

**Comment by masenf** [comment]
PR Submitted https://github.com/reflex-dev/reflex/pull/2801
